### DISCUSSION TITLE
Alternatives for npm to build the admin frontend

### DISCUSSION THREAD

**Discussion by stollr** [sulu/sulu#7160] [Q&A] [answered]
Our team is setting up the deployment process of our first Sulu project and because we have extended the admin frontend we need to build it manually. Today this is only possible with npm.

Looking at this [benchmark](https://p.datadoghq.eu/sb/d2wdprp9uki7gfks-c562c42f4dfd0ade4885690fa719c818?refresh_mode=sliding&tpl_var_npm%5B0%5D=%2A&tpl_var_pnpm%5B0%5D=%2A&tpl_var_yarn-classic%5B0%5D=%2A&tpl_var_yarn-modern%5B0%5D=%2A&tpl_var_yarn-nm%5B0%5D=%2A&tpl_var_yarn-pnpm%5B0%5D=no&from_ts=1686823743661&to_ts=1694599743661&live=true) npm is by far the slowest package manager in the benchmarks which are relevant for deployment (cache and lockfile).

I'd prefer yarn over npm. But maybe there are strong reasons for it. It would be interesting to know what the reason is for choosing npm.

**Comment by alexander-schranz** [comment]
The issues is more that `npm` >= 7 and `yarn` and `pnpm` are currently not supported because they don't handle `local dependencies` correctly. https://github.com/sulu/skeleton/issues/88

Currently only `npm@6` handles this correctly. The current version of `bun` currently works as tested last week but is not yet official supported as it is in an too early state that it maybe goes into the same direction as npm7, yarn or pnpm not installing local dependencies dependencies.

Why I understand because of git conflicts somebody don't want to commit the build, I still recommend that as nothing is faster then no build in the deployment and makes onboarding a developer into the project easier as they not even aslong not working in the frontend have to tackle that.

**Reply by stollr** [reply]
Okay, I understand the issue with the local dependencies now.
Today I have played a bit with yarn 3 and [portals](https://yarnpkg.com/protocol/portal) but I wasn't successful :-(

**Comment by alexander-schranz** [comment, marked as answer]
With the release of Sulu `2.6` we support a wider range of uptodate package manager tools:

 - npm 8, 9, 10
 - pnpm 8
 - bun 1

**Comment by stollr** [comment]
Thank you for the update!
